packages/@aws-cdk/aws-certificatemanager/lib/certificate.ts

```diff
@@ -135,3 +148,84 @@ class ImportedCertificate extends Construct implements ICertificate {
     return this.props;
   }
 }
+
+/**
+ * A certificate managed by AWS Certificate Manager.  Will be automatically
+ * validated using DNS validation against the specified Route 53 hosted zone.
+ */
+export class DNSValidatedCertificate extends Construct implements ICertificate  {
+  public readonly certificateArn: string;
+  private normalizedZoneName: string;
+  private hostedZoneId: string;
+  private domainName: string;
+
+  constructor(scope: Construct, id: string, props: DNSValidatedCertificateProps) {
+    super(scope, id);
+
+    this.domainName = props.domainName;
+    this.normalizedZoneName = props.hostedZone.zoneName;
+    // Remove trailing `.` from zone name
+    /* istanbul ignore if */
+    if (this.normalizedZoneName.endsWith('.')) {
+      this.normalizedZoneName = this.normalizedZoneName.substring(0, this.normalizedZoneName.length - 1);
+    }
+
+    // Remove any `/hostedzone/` prefix from the Hosted Zone ID
+    this.hostedZoneId = props.hostedZone.hostedZoneId;
+    /* istanbul ignore if */
+    if (this.hostedZoneId.startsWith('/hostedzone/')) {
+      this.hostedZoneId = this.hostedZoneId.split('/hostedzone/')[1];
+    }
+
+    const requestorFunction = new Function(this, 'CertificateRequestorFunction', {
+      code: new AssetCode(path.join(__dirname, 'request_handler')),
+      handler: 'index.certificateRequestHandler',
+      runtime: Runtime.NodeJS810,
+      timeout: 15 * 60 // 15 minutes
+    });
+    requestorFunction.addToRolePolicy(
+      new PolicyStatement()
+        .addActions('acm:RequestCertificate', 'acm:DescribeCertificate', 'acm:DeleteCertificate')
+        .addResource('*')
+    );
+    requestorFunction.addToRolePolicy(
+      new PolicyStatement()
+        .addActions('route53:GetChange')
+        .addResource('*')
+    );
+    requestorFunction.addToRolePolicy(
+      new PolicyStatement()
+        .addActions('route53:changeResourceRecordSets')
+        .addResource(`arn:aws:route53:::hostedzone/${this.hostedZoneId}`)
```

Comment:
Does the hosted zone not have an ARN accessor?

Reply:
Will fix

Reply:
No, it does not, see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-route53-hostedzone.html#w2ab1c21c10d195c18b9